Enhance keyword blocking to support content maps

pfefferle opened the pull request:
Updated `Moderation::activity_is_blocked` to check blocked keywords in all language variants by including `content_map`, `summary_map`, and `name_map` fields. Added corresponding unit test to verify keyword blocking in multilingual content.

See:

* https://www.w3.org/TR/activitystreams-vocabulary/#dfn-content
* https://www.w3.org/TR/activitystreams-vocabulary/#dfn-summary
* https://www.w3.org/TR/activitystreams-vocabulary/#dfn-name

## Proposed changes:
<!--- Explain what functional changes your PR includes -->

* Refactored keyword blocking logic to build a comprehensive content array from all available language variants
* Added support for ActivityStreams multilingual content maps (content_map, summary_map, name_map)
* Included unit test to verify keyword blocking works with multilingual content


### Other information:

- [X] Have you written new tests for your changes, if applicable?

## Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

* See unit-tests

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box below and supplying data. -->
<!-- It will trigger a GitHub workflow that will create and push the entry into the branch. -->

<!-- Due to org permissions, the job may fail for PRs crated from a fork under GitHub organizations. -->
<!-- In this case, you can create entry manually with `composer changelog:add` and push it into the branch. -->

<!-- If no changelog entry is required for this PR, please add the "Skip Changelog" label. -->

-   [X] Automatically create a changelog entry from the details below.

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [ ] Patch
-   [X] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [X] Added - for new features
-   [ ] Changed - for changes in existing functionality
-   [ ] Deprecated - for soon-to-be removed features
-   [ ] Removed - for now removed features
-   [ ] Fixed - for any bug fixes
-   [ ] Security - in case of vulnerabilities

#### Message

Moderation now checks blocked keywords across all language variants of the content, summary and name fields.

</details>

pfefferle committed:
Enhance keyword blocking to support content maps

Updated `Moderation::activity_is_blocked` to check blocked keywords in all language variants by including `content_map`, `summary_map`, and `name_map` fields. Added corresponding unit test to verify keyword blocking in multilingual content.

matticbot committed:
Add changelog

pfefferle committed:
no need to check for duplicates I think

pfefferle committed:
Align variable indentation in moderation and test files

Adjusted indentation for variable assignments in Moderation class and its corresponding test to improve code readability and maintain consistency.

obenland commented on includes/class-moderation.php:
```diff
@@ -242,11 +242,31 @@ private static function check_activity_against_blocks( $activity, $blocked_actor
+
+			if ( is_array( $object->get_name_map() ) ) {
+				$content_map = array_merge( $content_map, array_values( $object->get_name_map() ) );
+			}
```
```suggestion
			if ( \is_array( $object->get_content_map() ) ) {
				$content_map = \array_merge( $content_map, \array_values( $object->get_content_map() ) );
			}

			if ( \is_array( $object->get_summary_map() ) ) {
				$content_map = \array_merge( $content_map, \array_values( $object->get_summary_map() ) );
			}

			if ( \is_array( $object->get_name_map() ) ) {
				$content_map = \array_merge( $content_map, \array_values( $object->get_name_map() ) );
			}
```

obenland committed:
Add get_content_map, get_name_map, and get_summary_map methods

Documented new methods get_content_map, get_name_map, and get_summary_map in the class-generic-object.php docblock to reflect support for these properties.

pfefferle committed:
class-moderation.php aktualisieren

Co-authored-by: Konstantin Obenland <obenland@gmx.de>

pfefferle committed:
Merge branch 'trunk' into add/moderate_content_maps
